### DISCUSSION TITLE
Release 2022-12 and more on Eclipse alignments

### DISCUSSION THREAD

**Discussion by hazendaz** [revelc/formatter-maven-plugin#682] [General]
Eclipse hasn't done much in formatting area that I'm aware of last few releases but its probably ok to release the next cut.  There is new logic for handling resources more easily that is opt in which is a good improvement here.  Outside of outstanding work I'd love to get done, realistically I won't see myself using those updates probably until March.  So rather than back up into release after release of Eclipse, its probably best as a general rule to just release what we have once the main binaries are updated and if we get extra stuff great, but if not, that is ok too.

Issues I'm aware of in the current releases is that CSS doesn't handle proper end of line (return) at end of files.  Its not a big deal but may be more pronounced in case where 'resources' are now more easily used when opt in but probably not a show stopper to release this.

**Comment by ctubbsii** [comment]
> Eclipse hasn't done much in formatting area that I'm aware of last few releases

I think there was stuff changed to support newer Java language constructs, like the updated switch statements and records types.

> its probably best as a general rule to just release what we have once the main binaries are updated and if we get extra stuff great, but if not, that is ok too

That's my thinking, too. Is there a ticket for the existing issue for CSS? I'm not too concerned, as I only use this formatter for Java, anyway, but would be nice to link to it.

**Comment by hazendaz** [comment]
Haven't opened one yet for css issue.  Only have one know case of the issue and figured when I get to it I would just fix at same time.  However will open something up this weekend.

Sent from my Verizon, Samsung Galaxy smartphone
Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Reply by hazendaz** [reply]
I submitted fix for css issue.  After that is in, think we are good to release for 2022-12 release.

**Comment by ctubbsii** [comment]
I can release without #692 , but I tried to bump the xml-formatter to 0.3.0 after releasing that, and the XML formatting tests got NPEs. So, either something is broken in xml-formatter 0.3.0 or there's more work needed to migrate to it. I will release without that, and maybe we can get that working in the next release... perhaps a patch release if it's quick and easy.

**Comment by hazendaz** [comment]
Sure will look into it.

Sent from my Verizon, Samsung Galaxy smartphone
Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Reply by hazendaz** [reply]
xml 0.3.0 applied now.

**Reply by ctubbsii** [reply]
Is it worth doing a 2.22.1 or just wait until 2023-03 and include it in 2.23.0?

**Reply by hazendaz** [reply]
its fine to wait.
